checker/src/org/checkerframework/checker/genericeffects/GenericEffectExtension.java

```diff
@@ -0,0 +1,149 @@
+     *
+     * @param node The specific tree node that the developer wants to check.
+     * @return The effect of the specific tree node.
+     */
+    public Class<? extends Annotation> checkArrayAccess(ArrayAccessTree node) { return null; }
+    public Class<? extends Annotation> checkArrayType(ArrayTypeTree node) { return null; }
+    public Class<? extends Annotation> checkAssert(AssertTree node) { return null; }
+    public Class<? extends Annotation> checkAssignment(AssignmentTree node) { return null; }
+    public Class<? extends Annotation> checkBinary(BinaryTree node) { return null; }
+    public Class<? extends Annotation> checkBreak(BreakTree node) { return null; }
+    public Class<? extends Annotation> checkCase(CaseTree node) { return null; }
+    public Class<? extends Annotation> checkCatch(CatchTree node) { return null; }
+    public Class<? extends Annotation> checkCompoundAssignment(CompoundAssignmentTree node) { return null; }
+    public Class<? extends Annotation> checkConditionalExpression(ConditionalExpressionTree node) { return null; }
+    public Class<? extends Annotation> checkContinue(ContinueTree node) { return null; }
+    public Class<? extends Annotation> checkDoWhileLoop(DoWhileLoopTree node) { return null; }
+    public Class<? extends Annotation> checkEnhancedForLoop(EnhancedForLoopTree node) { return null; }
+    public Class<? extends Annotation> checkForLoop(ForLoopTree node) { return null; }
+    public Class<? extends Annotation> checkIf(IfTree node) { return null; }
+    public Class<? extends Annotation> checkInstanceOf(InstanceOfTree node) { return null; }
+    public Class<? extends Annotation> checkIntersectionType(IntersectionTypeTree node) { return null; }
+    public Class<? extends Annotation> checkLabeledStatement(LabeledStatementTree node) { return null; }
+    public Class<? extends Annotation> checkLiteral(LiteralTree node) { return null; }
+    public Class<? extends Annotation> checkNewArray(NewArrayTree node) { return null; }
+    public Class<? extends Annotation> checkPrimitiveType(PrimitiveTypeTree node) { return null; }
+    public Class<? extends Annotation> checkReturn(ReturnTree node) { return null; }
+    public Class<? extends Annotation> checkSwitch(SwitchTree node) { return null; }
+    public Class<? extends Annotation> checkSynchronized(SynchronizedTree node) { return null; }
+    public Class<? extends Annotation> checkThrow(ThrowTree node) { return null; }
+    public Class<? extends Annotation> checkTry(TryTree node) { return null; }
+    public Class<? extends Annotation> checkTypeCast(TypeCastTree node) { return null; }
+    public Class<? extends Annotation> checkUnary(UnaryTree ndoe) { return null; }
+    public Class<? extends Annotation> checkUnionType(UnionTypeTree node) { return null; }
+    public Class<? extends Annotation> checkWhileLoop(WhileLoopTree node) { return null; }
+    public Class<? extends Annotation> checkWildcard(WildcardTree node) { return null; }
```

Comment:
Rather than returning ```null```, these methods should throw ```java.lang.UnsupportedOperationException``` (https://docs.oracle.com/javase/7/docs/api/java/lang/UnsupportedOperationException.html).  They shouldn't be called at all if the corresponding ```doesXCheck()``` returns false.